java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/DatapathId.java

```diff
@@ -36,6 +39,19 @@ public static DatapathId of(byte[] bytes) {
         return new DatapathId(Longs.fromByteArray(bytes));
     }
 
+    /**
+     * Creates a {@link DatapathId} from a {@link MacAddress}. This factory
+     * method assumes that the {@link DatapathId} is composed of two zero bytes
+     * appended by the {@link MacAddress}'s 6 bytes.
+     * @param mac the {@link MacAddress} to create the {@link DatapathId} from
+     * @return a {@link DatapathId} derived from the supplied {@link MacAddress}
+     */
+    public static DatapathId of(@Nonnull MacAddress mac) {
+        byte[] zeroBytes = new byte[]{0,0};
```

Comment:
Same here - use getLong / of(long) to avoid ~3 byte array allocations/conversions.